File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Java/Employe.java
public abstract class Employe {
    protected String numeroSecuriteSociale;
    protected String nom;
    protected double salaire;
    protected String emploi;
    protected Employe superieur;

    public Employe(String numeroSecuriteSociale, String nom, double salaire, String emploi, Employe superieur) {
        this.numeroSecuriteSociale = numeroSecuriteSociale;
        this.nom = nom;
        this.salaire = salaire;
        this.emploi = emploi;
        this.superieur = superieur;
    }

    abstract public double calculerSalaire();

    public String afficherSuperieur() {
        return "Supérieur hiérarchique de " + nom + " : " + (superieur != null ? superieur.nom : "Aucun");
    }

    public String afficherSubordonnes() {
        return "\tSubordonnées :";
    }

    public String afficheEmploye() {
        return this.emploi + " : " + this.nom;
    }
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Cpp/Employe.cpp
#include "Employe.h"

Employe::Employe(string numeroSecuriteSociale, string nom, double salaire, string emploi, Employe * superieur) {
    this->numeroSecuriteSociale = numeroSecuriteSociale;
    this->nom = nom;
    this->salaire = salaire;
    this->emploi = emploi;
    this->superieur = superieur;
}

string Employe::afficherSuperieur() const {
    return "Supérieur hiérarchique de " + nom + " : " + (superieur != nullptr ? superieur->nom : "Aucun");
}

string Employe::afficherSubordonnes() const {
    return "\tSubordonnées :";
}

string Employe::afficheEmploye() const {
    return emploi + " : " + nom;
}